nntrainer/layers/layer.cpp

```diff
@@ -38,6 +38,11 @@ void Layer::setActivation(ActivationType acti) {
 }
 
 int Layer::setOptimizer(std::shared_ptr<Optimizer> opt) {
+  if (!opt) {
+    this->opt = opt;
```

Comment:
this is wrong for adam. If two layers share the optimizer, then the corresponding values inside the optimizer are also shared which creates issue. Thats why, a new optimizer is always created.

Reply:
Reverted.